standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java

```diff
@@ -1233,56 +1233,48 @@ private void create_database_core(RawStore ms, final Database db)
           madeExternalDir = true;
         }
       } else {
-        if (dbMgdPath != null) {
-          try {
-            // Since this may be done as random user (if doAs=true) he may not have access
-            // to the managed directory. We run this as an admin user
-            madeManagedDir = UserGroupInformation.getLoginUser().doAs(new PrivilegedExceptionAction<Boolean>() {
-              @Override public Boolean run() throws MetaException {
-                if (!wh.isDir(dbMgdPath)) {
-                  LOG.info("Creating database path in managed directory " + dbMgdPath);
-                  if (!wh.mkdirs(dbMgdPath)) {
-                    throw new MetaException("Unable to create database managed path " + dbMgdPath + ", failed to create database " + db.getName());
-                  }
-                  return true;
+        final Path mgdPath = dbMgdPath != null ? dbMgdPath : defaultDbMgdPath;
+        try {
```

Comment:
could you create a new method for creating both the managed dir and the external dir?